teamcity-snyk-security-plugin-agent/src/main/java/io/snyk/plugins/teamcity/agent/CommandExecutionAdapter.java

```diff
@@ -115,8 +117,15 @@ public void processFinished(int exitCode) {
 
           if (!snykApiResponse.success && nullIfEmpty(snykApiResponse.summary) != null) {
             String problem = format("%s known issues | %s", snykApiResponse.uniqueCount, snykApiResponse.summary);
-            BuildProblemData buildProblem = createBuildProblem(problem);
-            buildService.getLogger().logBuildProblem(buildProblem);
+
+            boolean containsFailOnIssues = buildService.getBuildRunnerContext().getRunnerParameters().containsKey(FAIL_ON_ISSUES);
```

Comment:
why to you need to get not only the value, but also whether the key exists?

Reply:
@yuliabaron, we have existing configurations without new parameter, it means for these configurations `failOnIssues` is not true => and we would change existing plugin behavior. To prevent it: I check that a new parameter exists (if no => our plugin was upgraded, but configuration was not updated, so we keep default "fail" behavior).

Reply:
Could you add it as a comment to the code please?

Reply:
done with 46223a7